packages/core/contracts/oracle/implementation/VotingV2.sol

```diff
@@ -365,19 +376,36 @@ contract VotingV2 is
         // on every commit and so it is not too computationally inefficient.
-        uint256 priceRequestIdsLength = priceRequestIds.length;
-        for (uint256 i = lastRequestIndexConsidered; i < priceRequestIdsLength; i = unsafe_inc(i)) {
-            if (deletedRequestJumpMapping[i] != 0) i = deletedRequestJumpMapping[i] + 1;
-            Request memory request = priceRequestIds[i];
-            PriceRequest storage priceRequest = priceRequests[request.requestId];
+        for (uint256 requestIndex = lastRequestIndexConsidered; requestIndex < priceRequestIds.length; requestIndex++) {
+            if (deletedRequests[requestIndex] != 0) requestIndex = deletedRequests[requestIndex] + 1;
+            if (requestIndex > priceRequestIds.length - 1) break; // This happens if the last element was a rolled vote.
+
+            PriceRequest storage priceRequest = priceRequests[priceRequestIds[requestIndex]];
             VoteInstance storage voteInstance = priceRequest.voteInstances[priceRequest.lastVotingRound];
 
-            // Cant slash this or any subsequent requests if the request is not settled. TODO: this has implications for
-            // rolled votes and should be considered closely.
-            if (_getRequestStatus(priceRequest, currentRoundId) != RequestStatus.Resolved) break;
-            uint256 stakedAtRound = rounds[request.roundId].cumulativeActiveStakeAtRound;
+            // If the request status is not resolved then: a) Either we are still in the current voting round, in which
+            // case break the loop and stop iterating (all subsequent requests will be in the same state by default) or
+            // b) we have gotten to a rolled vote in which case we need to update some internal trackers for this vote
+            // and set this within the deletedRequests mapping so the next time we hit this it is skipped.
+            if (_getRequestStatus(priceRequest, currentRoundId) != RequestStatus.Resolved) {
+                // If the request is not resolved and the lastVotingRound less than the current round then the vote
+                // must have been rolled. In this case, update the internal trackers for this vote.
+                if (priceRequest.lastVotingRound < currentRoundId) {
+                    priceRequest.lastVotingRound = currentRoundId;
+                    deletedRequests[requestIndex] = requestIndex;
+                    priceRequest.priceRequestIndex = priceRequestIds.length;
+                    priceRequestIds.push(priceRequestIds[requestIndex]);
+                    _updateCumulativeSlashingTrackers();
+                }
+                // Else, we are simply evaluating a request that is still actively being voted on. In this case, break as
+                // all subsequent requests within the array must be in the same state and cant have any slashing applied.
+                break;
```

Comment:
Question:
The comment says that there is an `else` condition, but if a rolled vote is found, the `break` command is always run. Is it what we want?